<launch>
  <arg name="image_detector_namespace" default="/detection/image_detector"/>
  <arg name="image_projector_namespace" default="/detection/image_projector"/>
  <!-- Camera Projector -->
  <node name="camera_projector" pkg="agv" type="camera_projector" output="screen">
    <param name="depth_image_topic"         value="/zed_node/depth/depth_registered"/>
    <param name="camera_objects_topic"      value="$(arg image_detector_namespace)/objects"/>
    <param name="output_objects_topic"      value="$(arg image_projector_namespace)/objects"/>
    <param name="target_frame"              value="/map"/>
    <param name="image_width"               value="1280"/>
    <param name="image_height"              value="720"/>
    <param name="image_fov_w"               value="85"/>
    <param name="image_fov_h"               value="54"/>
    <param name="depth_scale"               value="1.0"/>
    <param name="camera_height"             value="1.53"/>
    <param name="min_depth"                 value="0.4"/>
    <param name="max_depth"                 value="40.0"/>
  </node>

  <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="camera_projector_visualization"
    output="screen" ns="$(arg image_projector_namespace)"/>

</launch>
